http_handler.go

```diff
@@ -43,23 +46,35 @@ func Trace(handler http.Handler) http.Handler {
 // necessary.
 
 // In addition to tracing capabilities this function also provides graceful
-// shutdown of the server. When a SIGTERM signal is received by the server
-// it will stop accepting new connections but it will keep it running until
-// all the existing connections have been closed. It will kill the server
+// shutdown of the server. When a SIGTERM or SIGINT signal is received by the
+// server it will stop accepting new connections but it will keep it running
+// until all the existing connections have been closed. It will kill the server
 // after a default timeout of 25 seconds if there are still any pending
 // connections.
 
-func ListenAndServe(addr string, handler http.Handler) error {
+func ListenAndServe(addr string, handler http.Handler) {
```

Comment:
I think we should either make a function that returns an actually useful error, or a function that does not, but does not pretend to be ListenAndServe, not both.  

I think the cleanest would be to have a function with a different name and with this behaviour, because it will probably not be very useful to return errors from a function that has hijacked control of SIGINT/SIGTERM anyway. This would particularly make sense if we named it was made part of an actual service-kit package.